From 0c1090a26ce74b6e9507f707285309bf989b61e1 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 26 Aug 2020 14:59:42 -0700
Subject: [PATCH] drivers: hv: dxgkrnl: Initialize kmem in
 create_existing_sysmem

Avoids the following clang warnings:

drivers/hv/dxgkrnl/dxgvmbus.c:865:6: warning: variable 'kmem' is used
uninitialized whenever 'if' condition is true
[-Wsometimes-uninitialized]
        if (ret1 != npages) {
            ^~~~~~~~~~~~~~
drivers/hv/dxgkrnl/dxgvmbus.c:899:6: note: uninitialized use occurs here
        if (kmem)
            ^~~~
drivers/hv/dxgkrnl/dxgvmbus.c:865:2: note: remove the 'if' if its
condition is always false
        if (ret1 != npages) {
        ^~~~~~~~~~~~~~~~~~~~~
drivers/hv/dxgkrnl/dxgvmbus.c:858:6: warning: variable 'kmem' is used
uninitialized whenever 'if' condition is true
[-Wsometimes-uninitialized]
        if (dxgalloc->pages == NULL) {
            ^~~~~~~~~~~~~~~~~~~~~~~
drivers/hv/dxgkrnl/dxgvmbus.c:899:6: note: uninitialized use occurs here
        if (kmem)
            ^~~~
drivers/hv/dxgkrnl/dxgvmbus.c:858:2: note: remove the 'if' if its
condition is always false
        if (dxgalloc->pages == NULL) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/hv/dxgkrnl/dxgvmbus.c:840:12: note: initialize the variable
'kmem' to silence this warning
        void *kmem;
                  ^
                   = NULL
2 warnings generated.

Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
Signed-off-by: Diab Neiroukh <lazerl0rd@thezest.dev>
---
 drivers/hv/dxgkrnl/dxgvmbus.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hv/dxgkrnl/dxgvmbus.c b/drivers/hv/dxgkrnl/dxgvmbus.c
index d8d3d911b168..686c11c58ff7 100644
--- a/drivers/hv/dxgkrnl/dxgvmbus.c
+++ b/drivers/hv/dxgkrnl/dxgvmbus.c
@@ -837,7 +837,7 @@ int create_existing_sysmem(struct dxgdevice *device,
 			   const void *sysmem)
 {
 	int ret1 = 0;
-	void *kmem;
+	void *kmem = NULL;
 	int ret = 0;
 	struct dxgkvmb_command_setexistingsysmemstore set_store_command = { };
 	u64 alloc_size = host_alloc->allocation_size;
